src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -222,6 +222,11 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
                 }
             }
 
+            if (!entryPoint.HasBody)
```

Comment:
>if (!entryPoint.HasBody) [](start = 12, length = 24)

This check feels unnecessary. First, we call  GenerateMethodBody for synthesizedEntryPoint, not for entryPoint. Even though, it looks like they have the same value when synthesizedEntryPoint is not null, the code would be clearer if synthesizedEntryPoint was used. Second, it looks like synthesizedEntryPoint is always supposed to have a body. I'd suggest to replace the ```if``` with ```Debug.Assert(synthesizedEntryPoint == null || synthesizedEntryPoint.HasBody) ;```